### PR TITLE
fix: install fal and adapter local prereleases correctly

### PR DESCRIPTION
**Pull request by chamini2**
Was part of #727, taken out.

### CODE REVIEW

**Review by mederka — COMMENTED** [review]
Tests are running into this error:
```
No module named 'dbt.contracts.graph.parsed'
```

**Comment by chamini2** [comment]
> Tests are running into this error:
> 
> ```
> No module named 'dbt.contracts.graph.parsed'
> ```

We will see this go away after we merge #727 . Let's merge that first, we rebase and see these tests pass.

File: projects/adapter/src/dbt/adapters/fal_experimental/utils/environments.py [hunk 1]
```diff
@@ -311,15 +310,17 @@ def _version_is_prerelease(raw_version: str) -> bool:
     return package_version.is_prerelease
 
 
-def _get_adapter_root_path() -> Optional[Path]:
-    import dbt.adapters.fal as adapter
+def _get_project_root_path(pacakge: str) -> Path:
```
Comment (drochetti) [inline projects/adapter/src/dbt/adapters/fal_experimental/utils/environments.py:313]: small typo `s/pacakge/package`

```suggestion
def _get_project_root_path(package: str) -> Path:
```

File: projects/adapter/src/dbt/adapters/fal_experimental/utils/environments.py [hunk 2]
```diff
@@ -311,15 +310,17 @@ def _version_is_prerelease(raw_version: str) -> bool:
...
-    path = Path(adapter.__file__)
+    # If this is a development version, we'll install
+    # the current fal itself.
+    path = Path(fal.__file__)
     while path is not None:
         if (path.parent / ".git").exists():
             break
         path = path.parent
-    return (path / "adapter")
+    return path / pacakge
```
Comment (drochetti) [inline projects/adapter/src/dbt/adapters/fal_experimental/utils/environments.py:323]:
```suggestion
    return path / package
```

File: projects/fal/src/fal/packages/dependency_analysis.py [hunk 3]
```diff
@@ -10,15 +10,7 @@
...
-def _is_fal_pre_release() -> bool:
-    from dbt.semver import VersionSpecifier
-
-    raw_fal_version = importlib_metadata.version("fal")
-    fal_version = VersionSpecifier.from_version_string(raw_fal_version)
-    return fal_version.prerelease
-
-
-def _get_fal_root_path() -> Path:
+def _get_project_root_path(pacakge: str) -> Path:
```
Comment (drochetti) [inline projects/fal/src/fal/packages/dependency_analysis.py:13]:
```suggestion
def _get_project_root_path(package: str) -> Path:
```

File: projects/fal/src/fal/packages/dependency_analysis.py [hunk 4]
```diff
@@ -28,7 +20,7 @@ def _get_fal_root_path() -> Path:
         if (path.parent / ".git").exists():
             break
         path = path.parent
-    return (path / "fal")
+    return path / pacakge
```
Comment (drochetti) [inline projects/fal/src/fal/packages/dependency_analysis.py:23]:
```suggestion
    return path / package
```

**Review by drochetti — APPROVED** [review]
Minor comment about a typo, otherwise it lgtm! 🚀

**Pull request merged**